src/main/java/ch/njol/skript/effects/EffBlockUpdate.java

```diff
@@ -0,0 +1,89 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Name("BlockState - Update")
+@Description({
+	"Updates the blockstate of a block to cached state or selected block",
+	"`force`: Will force the update of the block",
+	"`without physics`: Does not send updates to surrounding blocks"
+})
+@Examples({
+	"set {_state} to blockstate of event-block",
+	"set event-block to air",
+	"wait 1 minute",
+	"force update {_state} without physics updates",
+	"",
+	"force update event-block as stone without physics updates"
+})
+@Since("INSERT VERSION")
+// Ported over from SkBee made by ShaneBee (Credits go to him)
+public class EffBlockUpdate extends Effect {
+
+	static {
+		Skript.registerEffect(EffBlockUpdate.class,
+			"[:force] update %blockstates% [physics:without (neighbour|physics) updates]",
```

Comment:
I'm fine with whatever, I was at the very least wanting to introduce BlockStates into Skript, just so that is there for "future proof". Fusezion did have a valid counter argument of how adding them now can be used to setup for the possibility of structures.

Reply:
Technically a blockstate is already implemented in said [structure pr](https://github.com/SkriptLang/Skript/pull/5222/) however the system we have there vs. the system we have here is different as for some reason it's making use of a dumb class skript created awhile back that really has no use in todays existence

Reply:
Blockstate is not implemented there, there's no blockstate classinfo. It's being presented as a block to the user.
I have thoughts about that and the possible confusion from it, but in that case it's already integrated into skript's systems and would natively support stuff like inventory, or setting it to different values, etc.